src/components/Carousel.jsx

```diff
@@ -0,0 +1,126 @@
+import React, { useState, useEffect } from "react";
+import {
+  Paper,
+  Typography,
+  Grid,
+  IconButton,
+  Card,
+  CardMedia,
+  CardContent,
+  Box,
+} from "@mui/material";
+import KeyboardArrowLeftIcon from "@mui/icons-material/KeyboardArrowLeft";
+import KeyboardArrowRightIcon from "@mui/icons-material/KeyboardArrowRight";
+import ModalDetails from "./ModalDetailes";
+
+const cardTextStyle = {
+  backgroundColor: "rgba(24, 20, 20, 0.7)",
+  color: "#fff",
+  padding: "8px",
+  position: "absolute",
+  bottom: "0",
+  left: "0",
+  right: "0",
+};
+
+function Carousel({ getFunction, title, genreId, apiFetched }) {
+  const [currentPage, setCurrentPage] = useState(1);
+  const [selectedItem, setSelectedItem] = useState(null);
+  const [itemsData, setItemsData] = useState([]);
+
+  const itemsPerPage = 8;
+  const dynamicNumberOfItems = 32;
+
+  useEffect(() => {
+    if (getFunction) {
+      const items =
+        genreId !== null
```

Comment:
stop with this syntax. it is wrong in so many levels!!!!
use
`!genereId` instead
In addition, useless if within if.
this check should be handled by this function, not its user.

Reply:
changed it